src/controllers/super-admin.js

```diff
@@ -33,47 +37,66 @@ const getSuperAdminsById = (req, res) => {
       error: true,
     });
   } else {
-    SuperAdmin.findById(id, 'email')
-      .then((superAdmin) => {
-        if (!superAdmin) {
-          res.status(400).json({
-            message: `This super admins with id: ${id} not exist`,
-            data: undefined,
-            error: true,
-          });
-        } else {
-          res.status(200).json({
-            message: 'Super admin found',
-            data: superAdmin,
-            error: false,
-          });
-        }
-      })
-      .catch((error) => res.status(400).json({
-        message: 'An error ocurred',
+    try {
+      const findByIdSuperAdmin = await SuperAdmin.findById(id, 'email');
+
+      if (!findByIdSuperAdmin) {
+        return res.status(400).json({
+          message: `This super admins with id: ${id} not exist`,
+          data: undefined,
+          error: true,
+        });
+      }
+      return res.status(200).json({
+        message: 'Super admin found',
+        data: findByIdSuperAdmin,
+        error: false,
+      });
+    } catch (error) {
+      return res.status(400).json({
+        message: error.message,
         data: undefined,
-        error,
-      }));
+        error: true,
+      });
+    }
   }
 };
 
-const createSuperAdmin = (req, res) => {
+// eslint-disable-next-line consistent-return
+const createSuperAdmin = async (req, res) => {
   const { email, password } = req.body;
 
-  SuperAdmin.create({ email, password })
-    .then((result) => res.status(201).json({
-      message: 'Succesfully created',
-      data: result,
-      error: false,
-    }))
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred',
+  const findEmail = await SuperAdmin.findOne({ email });
+
+  if (!findEmail) {
```

Comment:
You shouldn't put the try catch inside an If Else statement, I think it would be better to return the error before the try catch if a superadmin with that email already exists.

Reply:
Agreed, you could add the try before using the findOne method, since it could fail too.